ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Mock/Block/Praos.hs

```diff
@@ -0,0 +1,178 @@
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE RecordWildCards            #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE UndecidableInstances       #-}
+
+module Ouroboros.Consensus.Ledger.Mock.Block.Praos (
+    SimplePraosBlock
+  , SimplePraosHeader
+  , SimplePraosExt(..)
+  , SignedSimplePraos(..)
+  ) where
+
+import qualified Codec.CBOR.Decoding as CBOR
+import qualified Codec.CBOR.Encoding as CBOR
+import           Codec.Serialise (Serialise (..))
+import           GHC.Generics (Generic)
+
+import           Ouroboros.Consensus.Crypto.KES
+import           Ouroboros.Consensus.Ledger.Abstract
+import           Ouroboros.Consensus.Ledger.Mock.Address
+import           Ouroboros.Consensus.Ledger.Mock.Block
+import           Ouroboros.Consensus.Ledger.Mock.Forge
+import           Ouroboros.Consensus.Ledger.Mock.Stake
+import           Ouroboros.Consensus.Protocol.ExtNodeConfig
+import           Ouroboros.Consensus.Protocol.Praos
+import           Ouroboros.Consensus.Protocol.Signed
+import           Ouroboros.Consensus.Util (Empty)
+import           Ouroboros.Consensus.Util.Condense
+import qualified Ouroboros.Consensus.Util.SlotBounded as SB
+
+{-------------------------------------------------------------------------------
+  Instantiate the @ext@ to suit Praos
+-------------------------------------------------------------------------------}
+
+-- | Simple block extended with the fields required for Praos
+type SimplePraosBlock c c' = SimpleBlock c (SimplePraosExt c c')
+
+-- | Header for Proas
+type SimplePraosHeader c c' = SimpleHeader c (SimplePraosExt c c')
+
+-- | Block extension required for Praos
+newtype SimplePraosExt c c' = SimplePraosExt {
+    simplePraosExt :: PraosFields c' (SignedSimplePraos c c')
+  }
+  deriving (Generic, Condense, Show, Eq)
+
+-- | Part of the block that gets signed
+--
+-- TODO: Right now we sign all of the extra Praos fields. This may or may not
+-- be needed. <https://github.com/input-output-hk/cardano-ledger-specs/issues/530>
+-- Of course, this Praos is merely a proof of concept so it doesn't really
+-- matter either way; we include them here primarily to show that we can.
+data SignedSimplePraos c c' = SignedSimplePraos {
+      signedSimplePraos :: SimpleStdHeader c (SimplePraosExt c c')
+    , signedPraosFields :: PraosExtraFields c'
+    }
+
+-- | PBFT requires the ledger view; for the mock ledger, this is constant
```

Comment:
Wrong comment. Should explain why we need this mapping (UTxO is in terms of addresses, but stake is in terms of node IDs).

Reply:
Can refer to comment for `ProtocolLedgerView` instance.